platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc.py

```diff
@@ -0,0 +1,491 @@
+                raise ValueError("Json validation failed: invalid IP {}".format(header[ip]))
+
+    def validate_ipv6_header(self, header):
+        for ip in ['sip', 'dip']:
+            if ip in header and is_ip_valid(header[ip], IP_VERSION_IPV6) == False:
+                raise ValueError("Json validation failed: invalid IP {}".format(header[ip]))
+            
+    def validate_layer2_header(self, header):
+        for mac in ['smac', 'dmac']:
+            if mac in header and is_mac_valid(header[mac]) == False:
+                raise ValueError("Json validation failed: invalid mac {}".format(header[mac]))        
+        
+    def validate_header(self, header, is_outer_header=False):
+        ipv4_header = False
+        ipv6_header = False
+                
+        # Verify IPv4 and IPv6 headers do not co-exist in header
+        if 'ipv4' in header:
+            ipv4_header = True
+        if 'ipv6' in header:
+            ipv6_header = True
+        
+        if ipv4_header and ipv6_header:
+            raise ValueError("Json validation failed: IPv4 and IPv6 headers can not co-exist")
+
+        if ipv4_header:
+            # Verify valid IPs in header
+            self.validate_ipv4_header(header['ipv4'])
+
+            if is_outer_header:
+                if 'dip' not in header['ipv4']:
+                    raise ValueError("Json validation failed: destination IP is mandatory")
+
+                self.dst_ip = header['ipv4']['dip']
+                self.ip_version = IP_VERSION_IPV4
+
+            if 'tcp_udp' in header and 'proto' not in header['ipv4']:
+                raise ValueError("Json validation failed: transport protocol (proto) is mandatory when transport layer port exists")
+
+        elif ipv6_header:
+            self.validate_ipv6_header(header['ipv6'])
+
+            if is_outer_header:
+                if 'dip' not in header['ipv6']:
+                    raise ValueError("Json validation failed: destination IP is mandatory")
+
+                self.dst_ip = header['ipv6']['dip']
+                self.ip_version = IP_VERSION_IPV6
+
+            if 'tcp_udp' in header and 'next_header' not in header['ipv6']:
+                raise ValueError("Json validation failed: transport protocol (next_header) is mandatory when transport layer port exists")
+
+        # Verify valid macs in header
+        if header['layer2']:
+            self.validate_layer2_header(header['layer2'])    
+                   
+    def validate_outer_header(self):
+        try:
+            outer_header = self.packet['packet_info']['outer']
+        except KeyError:
+            raise ValueError("Json validation failed: outer header is mandatory")
+
+        self.validate_header(outer_header, is_outer_header=True)    
+        
+    def validate_inner_header(self):
+        inner_header = None
+        try:
+            inner_header = self.packet['packet_info']['inner']
```

Comment:
Instead we could do something like inner_header = self.packet['packet_info'].get('inner')  instead of doing try except. get will return None if there is no such key.

Reply:
I have changed the code to use get().